src/memory_region/local.rs

```diff
@@ -36,104 +44,134 @@ pub trait LocalMrReadAccess: MrAccess {
 
     /// Get the start pointer without lock
     ///
-    /// # Safety:
+    /// # Safety
     ///
-    /// make sure the mr is readable without cancel safety issue
-    ///
-    /// TODO: move unchecked methords to unsafe trait
+    /// Make sure the mr is readable without cancel safety issue
     #[inline]
     #[allow(clippy::as_conversions)]
-    fn as_ptr_unchecked(&self) -> *const u8 {
+    unsafe fn as_ptr_unchecked(&self) -> *const u8 {
         self.addr() as _
     }
 
     /// Get the memory region as slice until it is readable
     ///
     /// If this mr is being used in RDMA ops, the thread may be blocked
+    ///
+    /// Return `None` if the total size of this mr larger than `isize::MAX`
     #[inline]
     #[allow(clippy::as_conversions)]
-    fn as_slice(&self) -> MappedRwLockReadGuard<&[u8]> {
-        // SAFETY: unsoundness
-        // BUG: unsound public api
-        MappedRwLockReadGuard::map(self.as_ptr(), |ptr| unsafe {
-            slice::from_raw_parts(ptr, self.length())
-        })
+    fn as_slice(&self) -> Option<MappedRwLockReadGuard<&[u8]>> {
+        if self.length() > isize::MAX.cast() {
```

Comment:
The length can be checked when creating the memory region. 
The access trait is an unsafe trait now, so the length method can be trusted.

Reply:
Agree

Reply:
Check it here because other methods can be used even if its length is larger than `isize::MAX`.
But I do think it's a little redundant.